create-k8s-chained-sessions.sh

```diff
@@ -15,37 +15,43 @@ declare REGION='us-east-1'
 # Args:
 # 1: parent session id name from leapp
 # 2: sso role name to use for the parent session
-# 3: role name to use for the chained session
+# 3: scope of the IAM role ("panorama" or "eks"). used only for legacy reasons
+#    and can be removed when the keyword is removed from the associated IAM
+#    roles.
+# 4: name of the persona (e.g. admin, dev-writer, etc.) the new session is for
+# 5: version of kubernetes, used to determine the associated IAM role
 function createLeappSession {
     green_echo "creating chained session for $1 with role $3"
     parent_session_name=$1
     parent_role_name=$2
-    chained_role_name=$3
+    iam_role_scope=$3
+    persona_name=$4
+    k8s_version=$5
     # check if the parent session exists for the role. We do this because
-    # regular developers won't have the AWSAdministratorAccess role, so we
-    # don't want to create a chained session for them.
+    # not all users have access to all roles. We want to only create sessions
+    # for roles that people have access to.
     parent_session_id=$(leappSessionId "$parent_session_name" "$parent_role_name")
     if [[ -z "${parent_session_id}" ]]; then
         green_echo "    No parent session found for ${parent_session_name} with role ${parent_role_name}"
         return
     fi
 
-    chained_session_name="${parent_session_name}-${chained_role_name}"
+    chained_session_name="${parent_session_name}-${persona_name}"
 
     green_echo "    looking for existing session ${chained_session_name}"
-    chained_session_id=$(leappSessionId "$chained_session_name" "$chained_role_name")
+    chained_session_id=$(leappSessionId "$chained_session_name" "${persona_name}")
```

Comment:
This function is broken. The script should be idempotent, but after running it, the second time it still goes into the session creation logic and creates duplicate sessions.

You'll need to pass in the full role name or change the `jq` select function to work as a fuzzy filter.

Reply:
This has been fixed now. I retested by re-running the script; output is below (punycode deprecation errors removed manually for readability):
```
% ./create-k8s-chained-sessions.sh                                                                                                                                (~/Development/panorama/leapp-setup) panorama-k8s-playground-admin
creating chained session for panorama-k8s-playground with persona admin
    looking for existing session panorama-k8s-playground-admin
    existing session found
creating chained session for panorama-k8s-playground with persona dev-writer
    looking for existing session panorama-k8s-playground-dev-writer
    existing session found
creating chained session for panorama-k8s-playground with persona dev-reader
    looking for existing session panorama-k8s-playground-dev-reader
    existing session found
creating chained session for panorama-k8s-playground with persona data-science-tester
    looking for existing session panorama-k8s-playground-data-science-tester
    existing session found
creating chained session for panorama-k8s-playground-2 with persona admin
    looking for existing session panorama-k8s-playground-2-admin
    existing session found
creating chained session for panorama-k8s-playground-2 with persona dev-writer
    looking for existing session panorama-k8s-playground-2-dev-writer
    existing session found
creating chained session for panorama-k8s-playground-2 with persona dev-reader
    looking for existing session panorama-k8s-playground-2-dev-reader
    existing session found
creating chained session for panorama-k8s-playground-2 with persona data-science-tester
    looking for existing session panorama-k8s-playground-2-data-science-tester
    existing session found
creating chained session for panorama-k8s-staging with persona admin
    looking for existing session panorama-k8s-staging-admin
    existing session found
creating chained session for panorama-k8s-staging with persona dev-writer
    looking for existing session panorama-k8s-staging-dev-writer
    existing session found
creating chained session for panorama-k8s-staging with persona dev-reader
    looking for existing session panorama-k8s-staging-dev-reader
    existing session found
creating chained session for panorama-k8s-staging with persona data-science-tester
    looking for existing session panorama-k8s-staging-data-science-tester
    no existing session found; starting session for panorama-k8s-staging to get role arn
An error occurred (NoSuchEntity) when calling the GetRole operation: The role with name panorama-data-science-tester-1.24 cannot be found.
    creating new profile
    creating new session
creating chained session for panorama-k8s-production with persona admin
    looking for existing session panorama-k8s-production-admin
    existing session found
creating chained session for panorama-k8s-production with persona dev-writer
    looking for existing session panorama-k8s-production-dev-writer
    existing session found
creating chained session for panorama-k8s-production with persona dev-reader
    looking for existing session panorama-k8s-production-dev-reader
    existing session found
creating chained session for panorama-k8s-production with persona data-science-tester
An error occurred (NoSuchEntity) when calling the GetRole operation: The role with name panorama-data-science-tester-1.24 cannot be found.
```

Reply:
(fyi, you can get rid of the punycode output by changing your host installed node version. I opened [a ticket](https://github.com/Noovolari/leapp/issues/500) with the leapp team about it with the workaround I used described.